tests/robust/robust_fixtures.py

```diff
@@ -0,0 +1,208 @@
+import math
+from typing import Callable, Optional, TypeVar
+
+import pyro
+import pyro.distributions as dist
+import torch
+from pyro.nn import PyroModule
+
+from chirho.observational.handlers import condition
+from chirho.robust.internals.utils import ParamDict
+from chirho.robust.ops import Point
+
+pyro.settings.set(module_local_params=True)
+T = TypeVar("T")
+
+
+class SimpleModel(PyroModule):
+    def forward(self):
+        a = pyro.sample("a", dist.Normal(0, 1))
+        with pyro.plate("data", 3, dim=-1):
+            b = pyro.sample("b", dist.Normal(a, 1))
+            return pyro.sample("y", dist.Normal(a + b, 1))
+
+
+class SimpleGuide(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.loc_a = torch.nn.Parameter(torch.rand(()))
+        self.loc_b = torch.nn.Parameter(torch.rand((3,)))
+
+    def forward(self):
+        a = pyro.sample("a", dist.Normal(self.loc_a, 1))
+        with pyro.plate("data", 3, dim=-1):
+            b = pyro.sample("b", dist.Normal(self.loc_b, 1))
+            return {"a": a, "b": b}
+
+
+class GaussianModel(PyroModule):
+    def __init__(self, cov_mat: torch.Tensor):
+        super().__init__()
+        self.register_buffer("cov_mat", cov_mat)
+
+    def forward(self, loc):
+        pyro.sample(
+            "x", dist.MultivariateNormal(loc=loc, covariance_matrix=self.cov_mat)
+        )
+
+
+# Note: `gaussian_log_prob` is separate from the GaussianModel above because of upstream obstacles
+# in the interaction between `pyro.nn.PyroModule` and `torch.func`.
+# See https://github.com/BasisResearch/chirho/issues/393
+def gaussian_log_prob(params: ParamDict, data_point: Point[T], cov_mat) -> T:
+    with pyro.validation_enabled(False):
+        return dist.MultivariateNormal(
+            loc=params["loc"], covariance_matrix=cov_mat
+        ).log_prob(data_point["x"])
+
+
+class DataConditionedModel(PyroModule):
+    r"""
+    Helper class for conditioning on data.
+    """
+
+    def __init__(self, model: PyroModule):
+        super().__init__()
+        self.model = model
+
+    def forward(self, D: Point[T]):
```

Comment:
nit: type `D: Point[torch.Tensor]` since `DataConditionedModel` is not a `Generic` in `T`

Reply:
Done